### DISCUSSION TITLE
Open small native App inside RN App that uses react navigation

### DISCUSSION THREAD

**Discussion by wilsolutions** [react-navigation/react-navigation#9978] [Q&A] [answered]
Hi, pls, how can I load a small iOS native App inside of an existing ones that uses react navigation?

I have tried creating a RN module that returns a view. 
In the view I use RCTPresentedViewController to get the View Controller, then I add the native app view controller as child.

```
*** Terminating app due to uncaught exception 'UIViewControllerHierarchyInconsistency', reason: 'child view controller:<react_native_demolib.TodosViewController: 0x7fb39a09c630> should have parent view controller:<RNSScreen: 0x7fb39a17b060> but actual parent is:<UIViewController: 0x7fb39a016350>'
terminating with uncaught exception of type NSException
```

So, using navigation v5 I was able to get it working setting enableScreens(false); but with v6 it is not working.

Update, I'm using:
```
    "@react-navigation/bottom-tabs": "6.0.5",
    "@react-navigation/drawer": "6.1.4",
    "@react-navigation/material-top-tabs": "6.0.2",
    "@react-navigation/native": "6.0.2",
    "@react-navigation/routers": "6.0.1",
    "@react-navigation/stack": "6.0.7",
```

ty

**Comment by satya164** [comment, marked as answer]
Are you using `@react-navigation/stack` or `@react-navigation/native-stack`?

**Reply by wilsolutions** [reply]
Hi @satya164 thank you, I forgot to mention that, I'm using "@react-navigation/stack": "6.0.7".

**Reply by satya164** [reply]
@wilsolutions `enableScreens(false)` should work, but you can also specify `detachInactiveScreens={false}` on the navigator

**Reply by wilsolutions** [reply]
Awesome! 🥳

The `detachInactiveScreens={false}` on the navigator got it working! Loved! 

Thank you!
